gnocchi/json.py

```diff
@@ -30,8 +30,7 @@ def to_primitive(obj):
     if isinstance(obj, datetime.datetime):
         return obj.isoformat()
     if isinstance(obj, numpy.datetime64):
-        # Do not include nanoseconds if null
-        return str(obj).rpartition(".000000000")[0] + "+00:00"
+        return numpy.datetime_as_string(obj, unit='s') + "+00:00"
```

Comment:
don't think this is wrong but can you confirm the issue here? is this a separate bug you're fixing?

Reply:
it is another bug that we discovered while fixing this issue of slowness. The date manipulation here is quite odd, which was causing some weird outputs if the nanoseconds part of the string had some value different from `.000000000`;  we just changed that to a better approach.

Reply:
i see. have a slight preference to have this in a separate PR/commit.

Reply:
This is also something that we found while troubleshooting this issue here.